src/main/java/core/basesyntax/Circle.java

```diff
@@ -0,0 +1,24 @@
+package core.basesyntax;
+
+public class Circle extends Figure {
+    private final double radius;
+    private final ColorSupplier color;
+
+    public Circle(ColorSupplier color, double radius) {
```

Comment:
Creating a figure, don't pass expressions in the constructor. You should pass a Color object, not a ColorSupplier. Create a Color object outside of the constructor and pass it on for better code readability.